client/packages/core/src/Reactor.js

```diff
@@ -431,33 +431,72 @@ export default class Reactor {
       case 'refresh-ok':
         const { computations, attrs } = msg;
         this._setAttrs(attrs);
+
+        const mutations = this._rewriteMutations(
+          attrs,
+          this.pendingMutations.currentValue,
+        );
+
+        const processedTxId = msg['processed-tx-id'];
+
         const updates = computations.map((x) => {
           const q = x['instaql-query'];
           const result = x['instaql-result'];
           const hash = weakHash(q);
           const triples = extractTriples(result);
-          const store = s.createStore(
+          let store = s.createStore(
```

Comment:
One PR you may find interesting: 

https://github.com/instantdb/instant/pull/613/files

Previously we did something similar and cleaned up mutations in refresh. I think at that point we were too eager about when we cleaned pending mutations. I _think_ your current solution is more robust, but wanted to share the previous PR, in case it helps with context